hw/drivers/adc/adc_nrf52/src/adc_nrf52.c

```diff
@@ -135,19 +135,31 @@ static int
 nrf52_adc_close(struct os_dev *odev)
 {
     struct adc_dev *dev;
+    int rc = 0;
+    int unlock = 0;
 
     dev = (struct adc_dev *) odev;
 
-    nrfx_saadc_uninit();
+    if (os_started()) {
+        rc = os_mutex_pend(&dev->ad_lock, OS_TIMEOUT_NEVER);
+        if (rc != OS_OK) {
+            goto err;
+        }
+        unlock = 1;
+    }
+    if (dev->ad_dev.od_open_ref == 1) {
```

Comment:
Where does od_open_ref get decremented?

Reply:
It is in os_dev.c . Actually I'm not sure why os_dev.c always calls od_open() and od_close() no matter on reference counter.

Maybe we could change a bit logic here and  make  os_dev.c to call od_open() only once when reference count is 0 and calls od_close() only once when reference count drops back to 0. In this way, driver does not have to  check od_open_ref.

Reply:
Yeah, I think I had made a similar comment on the dev list. That changes seems reasonable to me.